ssv/spectest/tests/runner/pre_consensus_justifications/invalid_data.go

```diff
@@ -0,0 +1,110 @@
+					testingutils.SSVMsgAggregator(invalidateMsgDataF(testingutils.AggregatorMsgID), nil),
+				},
+				PostDutyRunnerStateRoot: "db1b416873d19be76cddc92ded0d442ba0e642514973b5dfec45f587c6ffde15",
+				OutputMessages: []*types.SignedPartialSignatureMessage{
+					testingutils.PreConsensusSelectionProofMsg(ks.Shares[1], ks.Shares[1], 1, 1), // broadcasts when starting a new duty
+				},
+				ExpectedError: expectedErr,
+			},
+			{
+				Name:   "randao",
+				Runner: decideFirstHeight(testingutils.ProposerRunner(ks)),
+				Duty:   &testingutils.TestingProposerDuty,
+				Messages: []*types.SSVMessage{
+					testingutils.SSVMsgProposer(invalidateMsgDataF(testingutils.ProposerMsgID), nil),
+				},
+				PostDutyRunnerStateRoot: "2754fc7ced14fb15f3f18556bb6b837620287cbbfbf908abafa5a0533fc4bc5f",
+				OutputMessages: []*types.SignedPartialSignatureMessage{
+					testingutils.PreConsensusRandaoMsg(ks.Shares[1], 1), // broadcasts when starting a new duty
+				},
+				ExpectedError: expectedErr,
+			},
+			{
+				Name:   "randao (blinded block)",
+				Runner: decideFirstHeight(testingutils.ProposerBlindedBlockRunner(ks)),
+				Duty:   &testingutils.TestingProposerDuty,
+				Messages: []*types.SSVMessage{
+					testingutils.SSVMsgProposer(invalidateMsgDataF(testingutils.ProposerMsgID), nil),
+				},
+				PostDutyRunnerStateRoot: "6bd59da9f817b8e40112e58231e36738b9d021db4416c9eeec1dd0236a5362e2",
+				OutputMessages: []*types.SignedPartialSignatureMessage{
+					testingutils.PreConsensusRandaoMsg(ks.Shares[1], 1), // broadcasts when starting a new duty
+				},
+				ExpectedError: expectedErr,
+			},
+			{
+
+				Name:   "attester",
+				Runner: decideFirstHeight(testingutils.AttesterRunner(ks)),
+				Duty:   &testingutils.TestingAttesterDuty,
+				Messages: []*types.SSVMessage{
+					testingutils.SSVMsgAttester(invalidateMsgDataF(testingutils.AttesterMsgID), nil),
+				},
+				PostDutyRunnerStateRoot: "81cb7b1d3ea3087d49f9773b3a2b75a87b901e50427d237f2a10c0e1904e7684",
+				OutputMessages:          []*types.SignedPartialSignatureMessage{},
+				ExpectedError:           "failed processing consensus message: could not process msg: invalid signed message: proposal not justified: proposal fullData invalid: failed decoding consensus data: incorrect offset",
+			},
+			{
+				Name:   "sync committee",
+				Runner: decideFirstHeight(testingutils.SyncCommitteeRunner(ks)),
+				Duty:   &testingutils.TestingSyncCommitteeDuty,
+				Messages: []*types.SSVMessage{
+					testingutils.SSVMsgSyncCommittee(invalidateMsgDataF(testingutils.SyncCommitteeMsgID), nil),
+				},
+				PostDutyRunnerStateRoot: "38592232077cd45709a7c6cfdd20c9d899af9d79bc750add3c4b8f2b6794cb34",
+				OutputMessages:          []*types.SignedPartialSignatureMessage{},
+				ExpectedError:           "failed processing consensus message: could not process msg: invalid signed message: proposal not justified: proposal fullData invalid: failed decoding consensus data: incorrect offset",
```

Comment:
I just don't see how those tests relate to preconsensus justifications

Reply:
pre-consensus justifications are in the ConsensusData object, if you can't decode it than the justification is invalid